src/main/java/com/feedzai/commons/sql/abstraction/engine/AbstractTranslator.java

```diff
@@ -373,6 +375,48 @@ public String translate(final Case aCase) {
                              elseString);
     }
 
+    /**
+     * Translates Cast.
+     *
+     * @param cast a cast expression.
+     * @return cast translation.
+     */
+    public String translate(final Cast cast) {
+        // PDB does not support cast to JSON, CLOB and BLOB.
+        // An exception will be thrown when trying to translate the data type.
+
+        inject(cast.getExpression());
+        return String.format("CAST(%s AS %s)",
+                cast.getExpression().translate(),
+                this.translate(cast.getType()));
+    }
+
+    /**
+     * Translates the expression and tries to return its boolean value.
+     * If it can't, it returns null.
+     *
+     * @param expression the expression to translate the boolean value.
+     * @return the boolean value if possible, null otherwise.
+     */
+    protected String tryBooleanTranslate(final Expression expression) {
+
+        // Have the translation easier to match the regex below.
+        final String translation = expression.translate()
+                .replaceAll("'", "")
```

Comment:
I am not comfortable with this. Need to think.

Reply:
We should discuss this offline.